pkg/apis/cluster/validation/validation.go

```diff
@@ -90,6 +92,10 @@ func ValidateClusterSpec(spec *api.ClusterSpec, fldPath *field.Path) field.Error
 		allErrs = append(allErrs, ValidateClusterProxyURL(fldPath.Child("proxyURL"), spec.ProxyURL)...)
 	}
 
+	allErrs = append(allErrs, validateClusterForSelector(fldPath.Child("provider"), spec.Provider))
+	allErrs = append(allErrs, validateClusterForSelector(fldPath.Child("region"), spec.Region))
+	allErrs = append(allErrs, validateClusterForSelector(fldPath.Child("zone"), spec.Zone))
```

Comment:
I'm afraid this will append a `nil` error to `allErrs`.

Reply:
CI failed again. I guess this is the reason.

Reply:
Sorry for I didn't clear it at my previous comment: https://github.com/karmada-io/karmada/pull/2849#discussion_r1030088832

The `validateClusterForSelector` can return a error slice, so that we can append it outside. append(x, y`...`).